crates/matrix-sdk-ui/src/timeline/inner/mod.rs

```diff
@@ -319,7 +321,10 @@ impl<P: RoomDataProvider> TimelineInner<P> {
             }
 
             TimelineFocusData::PinnedEvents { loader } => {
-                let loaded_events = loader.load_events().await.map_err(Error::PinnedEventsError)?;
+                let loaded_events = loader
+                    .load_events(pinned_event_cache)
```

Comment:
Unsure if this is doable but can't we pass the cache directly to the loader when constructing it?

Reply:
Maybe we could make a clone since it'll anyway reuse the `Arc` inside it, but I wanted to keep this working with similar APIs to those of `EventCache`,  and that one is passed around.